opensoundscape/torch/models/cnn.py

```diff
@@ -454,114 +463,30 @@ def train(
                     [f.write(p + "\n") for p in bad_paths]
             warnings.warn(msg)
 
-    def save(self, path=None, save_weights=True, save_optimizer=True, extras={}):
-        """save model with weights (default location is self.save_path)
-
-        Args:
-            path: destination for saved model. if None, uses self.save_path
-            save_weights: if False, only save metadata/metrics [default: True]
-            save_optimizer: if False, don't save self.optim.state_dict()
-            extras: arbitrary dictionary of things to save, eg valid-preds
-        """
-
-        if path is None:
-            path = f"{self.save_path}/epoch-{self.current_epoch}.model"
-        path = Path(path)
-        os.makedirs(path.parent, exist_ok=True)
-
-        # add items to save into a dictionary
-        model_dict = {
-            "model": self.name,
-            "classes": self.classes,
-            "epoch": self.current_epoch,
-            "valid_metrics": self.valid_metrics,
-            "train_metrics": self.train_metrics,
-            "loss_hist": self.loss_hist,
-            "lr_update_interval": self.lr_update_interval,
-            "lr_cooling_factor": self.lr_cooling_factor,
-            "optimizer_params": self.optimizer_params,
-            "single_target": self.single_target,
-        }
-        if save_weights:
-            model_dict.update({"model_state_dict": self.network.state_dict()})
-        if save_optimizer:
-            if self.opt_net is None:
-                self.opt_net = self._init_optimizer()
-            model_dict.update({"optimizer_state_dict": self.opt_net.state_dict()})
-
-        # user can provide an arbitrary dictionary of extra things to save
-        model_dict.update(extras)
+    def save(model, path, save_datasets=True):
```

Comment:
There's no self argument in this method. Is this right?

Reply:
hmm this is very wrong, good catch. It only works because it should say "self" where "model" is - self is getting passed to the "model" argument which works as intended but is misleading (the function was written outside of the class then moved into the class).